protocol/test/forge/v2/circuitBreaker/TempleCircuitBreakerProxy.t.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { TempleTest } from "../../TempleTest.sol";
+
+import { TempleCircuitBreakerAllUsersPerPeriod } from "contracts/v2/circuitBreaker/TempleCircuitBreakerAllUsersPerPeriod.sol";
+import { TempleCircuitBreakerProxy } from "contracts/v2/circuitBreaker/TempleCircuitBreakerProxy.sol";
+import { FakeERC20 } from "contracts/fakes/FakeERC20.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+
+/* solhint-disable func-name-mixedcase, not-rely-on-time */
+contract TempleCircuitBreakerProxyTest is TempleTest {
+    TempleCircuitBreakerAllUsersPerPeriod public templeCircuitBreaker;
+    TempleCircuitBreakerAllUsersPerPeriod public daiCircuitBreaker;
+    TempleCircuitBreakerProxy public circuitBreakerProxy;
+
+    FakeERC20 public daiToken;
+    FakeERC20 public templeToken;
+    address caller = makeAddr("caller");
+
+    bytes32 public constant EXTERNAL_ALL_USERS = keccak256("EXTERNAL_USER");
+
+    event CircuitBreakerSet(bytes32 indexed identifier, address indexed token, address circuitBreaker);
+    event IdentifierForCallerSet(address indexed caller, string identifierString, bytes32 identifier);
+
+    function setUp() public {
+        daiToken = new FakeERC20("DAI Token", "DAI", executor, 500_000e18);
+        templeToken = new FakeERC20("Temple Token", "TEMPLE", executor, 500_000e18);
+        circuitBreakerProxy = new TempleCircuitBreakerProxy(rescuer, executor);
+        templeCircuitBreaker = new TempleCircuitBreakerAllUsersPerPeriod(rescuer, executor, 26 hours, 13, 1_000e18);
+        daiCircuitBreaker = new TempleCircuitBreakerAllUsersPerPeriod(rescuer, executor, 26 hours, 13, 1_000e18);
+    }
+
+    function test_initialisation() public {
+        assertEq(circuitBreakerProxy.executor(), executor);
+        assertEq(circuitBreakerProxy.rescuer(), rescuer);
+        bytes32[] memory ids = circuitBreakerProxy.identifiers();
+        assertEq(ids.length, 0);
+    }
+
+    function test_access_setIdentifierForCaller() public {
+        expectElevatedAccess();
+        circuitBreakerProxy.setIdentifierForCaller(address(0), "EXTERNAL_USER");
+    }
+
+    function test_access_setCircuitBreaker() public {
+        expectElevatedAccess();
+        circuitBreakerProxy.setCircuitBreaker(EXTERNAL_ALL_USERS, address(templeToken), address(templeCircuitBreaker));
+    }
+
+    function test_access_preCheck() public {
+        // Alice isn't mapped - so this will revert
+        vm.startPrank(alice);
+        vm.expectRevert();
```

Comment:
nit: add specific custom error

Reply:
I intentionally didn't do this as it impacts gas on every call (see comment in `TempleCircuitBreakerProxy::preCheck()`). It's just a mapping/setup issue so won't affect users